src/Components/Components/src/Services/UriHelperBase.cs

```diff
@@ -73,8 +73,26 @@ public void NavigateTo(string uri, bool forceLoad)
         /// Called to initialize BaseURI and current URI before those values the first time.
         /// Override this method to dynamically calculate the those values.
         /// </summary>
-        protected virtual void InitializeState()
+        public virtual void InitializeState(string uriAbsolute, string baseUriAbsolute)
```

Comment:
This seems wrong. to have public.

Reply:
Needs to be public so that we can call it from MVC

Reply:
Isn't this an argument for making something `public` in whatever subclass of `UriHelperBase` is used by MVC, rather than making it public on the base class? Or is that not practical for some reason?

Reply:
* We agreed that MVC was going to use a separate codepath to render components statically (not dependent on IComponentPrerenderer).
* We agreed that there were going to be no references from MVC to Components.Server and viceversa.
* This component needs to initialize the UriHelper so that requests can get routed properly. In order to do so, InitializeState needs to be public so that MVC can call into it (see line below).
https://github.com/aspnet/AspNetCore/pull/7954/files#diff-bc676323fcbc8b60ac8dc402fbb55f8aR49